prow/kube/prowjob.go

```diff
@@ -115,6 +115,20 @@ type ProwJobSpec struct {
 	// of this job that can run in parallel at once
 	MaxConcurrency int `json:"max_concurrency,omitempty"`
 
+	// Decorate determines if the user-provided PodSpec
```

Comment:
Group TimeoutMinutes with these fields.

Reply:
Thoughts on keeping these fields as is vs grouping into a struct similar to UtilityConfig? Top-level spec fields should be as reusable across agents as possible.

Reply:
That's the sort of bike-shed I want to have. I think the current design of one spec for many controllers is bad and I'm not sure we want to stick with it forever but if we want to move these, sure, I don't feel too strongly

Reply:
I don't think it's bad, we just don't do unions correctly.

Reply:
Btw there is a proposal to add first-class support for union in k8s

https://github.com/kubernetes/community/pull/388